File: backend/src/main/java/hu/szakdoga/backend/timetable/data/dto/TeacherDTO.java
package hu.szakdoga.backend.timetable.data.dto;

import javax.persistence.Column;

public class TeacherDTO {
    public final Long id;
    public final String name;
    public final String webpage;
    public final String email;
    public final Long userId;

    public final String office;

    public final String moreInformation;

    public TeacherDTO(Long id, String name, String webpage, String email, Long userId, String office, String moreInformation) {
        this.id = id;
        this.name = name;
        this.webpage = webpage;
        this.email = email;
        this.userId = userId;
        this.office = office;
        this.moreInformation = moreInformation;
    }
}

File: frontend/src/app/shared/model/timetable/dto/teacher.dto.ts
export class TeacherDto {
    public readonly id: number | null;
    public readonly name: string;
    public readonly webpage: string;
    public readonly email: string;
    public readonly userId: number;
    public readonly office: string;
    public readonly moreInformation: string;

    constructor(name: string, webpage: string, email: string, userId: number, office: string, moreInformation: string, id: number | null = null) {
        this.id = id;
        this.name = name;
        this.webpage = webpage;
        this.email = email;
        this.userId = userId;
        this.office = office;
        this.moreInformation = moreInformation;
    }
}
